lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/summary/APrioriSummarizer.java

```diff
@@ -50,16 +56,19 @@ public void process(DataFrame input) throws Exception {
         d = attributes.size();
 
         // Marking Outliers
-        boolean[] flag = new boolean[n];
-        double[] outlierCol = input.getDoubleColumnByName(outlierColumn);
-        numOutliers = 0;
+        outlierCol = input.getDoubleColumnByName(outlierColumn);
+        if (input.hasColumn(countColumn)) {
+            countCol = input.getDoubleColumnByName(countColumn);
+            hasCountCol = true;
+        }
+        numTotal = 0;
+        double numOutliersExact = 0.0;
         for (int i = 0; i < n; i++) {
-            flag[i] = predicate.test(outlierCol[i]);
-            if (flag[i]) {
-                numOutliers++;
-            }
+            numTotal += hasCountCol ? countCol[i] : 1;
```

Comment:
cool, I changed this one, but there are two other similar pieces of code in countSingles and countSet where it's harder to move the condition outside the loop without adding a lot of redundant code, so I left those unchanged.